diff --git a/data/autosuspend.hwdb b/data/autosuspend.hwdb
index 6c15fb9..e80d7b8 100644
--- a/data/autosuspend.hwdb
+++ b/data/autosuspend.hwdb
@@ -204,6 +204,7 @@ usb:v10A5pC844*
 # Supported by libfprint driver goodixmoc
 usb:v27C6p5840*
 usb:v27C6p6014*
+usb:v27C6p6090*
 usb:v27C6p6092*
 usb:v27C6p6094*
 usb:v27C6p609A*
diff --git a/libfprint/drivers/goodixmoc/goodix.c b/libfprint/drivers/goodixmoc/goodix.c
index 976675b..70bd373 100644
--- a/libfprint/drivers/goodixmoc/goodix.c
+++ b/libfprint/drivers/goodixmoc/goodix.c
@@ -1377,6 +1377,7 @@ gx_fp_probe (FpDevice *device)
     case 0x60A2:
     case 0x60A4:
     case 0x6014:
+    case 0x6090:
     case 0x6092:
     case 0x6094:
     case 0x609A:
@@ -1626,6 +1627,7 @@ fpi_device_goodixmoc_init (FpiDeviceGoodixMoc *self)
 static const FpIdEntry id_table[] = {
   { .vid = 0x27c6,  .pid = 0x5840,  },
   { .vid = 0x27c6,  .pid = 0x6014,  },
+  { .vid = 0x27c6,  .pid = 0x6090,  },
   { .vid = 0x27c6,  .pid = 0x6092,  },
   { .vid = 0x27c6,  .pid = 0x6094,  },
   { .vid = 0x27c6,  .pid = 0x609A,  },
