diff --git a/package.json b/package.json
index 1dee25fea..f2e02019c 100644
--- a/package.json
+++ b/package.json
@@ -51,12 +51,6 @@
       "description": "Search and open recent projects from VS Code",
       "mode": "view"
     },
-    {
-      "name": "open-with-vscode",
-      "title": "Open with Visual Studio Code",
-      "description": "Open the currently selected Finder item with VS Code",
-      "mode": "no-view"
-    },
     {
       "name": "open-new-window",
       "title": "Open New Window",
@@ -196,7 +190,6 @@
       "type": "textfield",
       "title": "Git Integration Color",
       "required": false,
-      "label": "Git Branch Color",
       "default": "",
       "description": "Hexadecimal color code for Git branch tag (e.g., #00FF00). Leave empty to use default green"
     }
diff --git a/src/db.ts b/src/db.ts
index bcbffb937..8e13dd616 100644
--- a/src/db.ts
+++ b/src/db.ts
@@ -88,7 +88,7 @@ function getPath() {
   if (isWin) {
     return `${homedir()}\\AppData\\Roaming\\${build}\\User\\globalStorage\\state.vscdb`;
   }
-  return `${homedir()}/Library/Application Support/${build}/User/globalStorage/state.vscdb`;
+  return `${homedir()}/.config/Code/User/globalStorage/state.vscdb`;
 }
 
 async function saveEntries(entries: EntryLike[]) {
diff --git a/src/index.tsx b/src/index.tsx
index d92cbca60..8321feb7f 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -1,4 +1,15 @@
-import { Action, ActionPanel, Color, Grid, Icon, open, openExtensionPreferences, showToast, Toast } from "@raycast/api";
+import {
+  Action,
+  ActionPanel,
+  Color,
+  Grid,
+  Icon,
+  open,
+  openExtensionPreferences,
+  showToast,
+  Toast,
+  closeMainWindow,
+} from "@raycast/api";
 import { usePromise } from "@raycast/utils";
 import { basename, dirname } from "path";
 import { useEffect, useState } from "react";
@@ -187,32 +198,19 @@ function LocalItem(
   };
 
   const getAction = (revert = false) => {
-    if (isWin) {
-      return () => {
-        const cliFilename = getVSCodeCLIFilename();
-        const fp = fileURLToPath(props.uri);
-        runExec([cliFilename, fp], (error) => {
-          if (error) {
-            console.error(`Error opening file: ${error}`);
-            showToast(Toast.Style.Failure, `Failed to open file: ${error}`);
-            return;
-          }
-        });
-      };
-    }
     return () => {
-      if (closeOtherWindows !== revert) {
-        runAppleScriptSync(`
-        tell application "System Events"
-          tell process "${build}"
-            repeat while window 1 exists
-              click button 1 of window 1
-            end repeat
-          end tell
-        end tell
-        `);
-      }
-      open(props.uri, bundleIdentifier);
+      closeMainWindow();
+      const cliFilename = getVSCodeCLIFilename();
+      const fp = fileURLToPath(props.uri);
+      runExec([cliFilename, fp], (error) => {
+        if (error) {
+          console.error(`Error opening file: ${error}`);
+          showToast(Toast.Style.Failure, `Failed to open file: ${error}`);
+          return;
+        } else {
+          showToast(Toast.Style.Success, `Opened in ${build}`);
+        }
+      });
     };
   };
 
@@ -238,8 +236,8 @@ function LocalItem(
       id={props.pinned ? path : undefined}
       title={name}
       subtitle={displaySubtitle}
-      icon={{ fileIcon: path }}
-      content={{ fileIcon: path }}
+      icon={Icon.Folder}
+      content={Icon.Folder}
       keywords={keywords}
       accessories={accessories}
       actions={
@@ -257,7 +255,6 @@ function LocalItem(
               onAction={getAction(true)}
               shortcut={{ modifiers: ["cmd", "shift"], key: "enter" }}
             />
-            <Action.OpenWith path={path} shortcut={{ modifiers: ["cmd"], key: "o" }} />
             {isFolderEntry(props.entry) && terminalApp && (
               <Action
                 title={`Open with ${terminalApp.name}`}
diff --git a/src/lib/vscode.ts b/src/lib/vscode.ts
index ed8cf0da8..c5f8b4c46 100644
--- a/src/lib/vscode.ts
+++ b/src/lib/vscode.ts
@@ -66,17 +66,7 @@ function getNLSVariable(text: string | undefined): string | undefined {
   }
 }
 const cliPaths: Record<string, string> = {
-  Antigravity: "/Applications/Antigravity.app/Contents/Resources/app/bin/antigravity",
-  Code: "/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code",
-  "Code - Insiders": "/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/bin/code",
-  Cursor: "/Applications/Cursor.app/Contents/Resources/app/bin/cursor", // it also has code, which is an alias
-  Kiro: "/Applications/Kiro.app/Contents/Resources/app/bin/kiro",
-  Positron: "/Applications/Positron.app/Contents/Resources/app/bin/code",
-  Trae: "/Applications/Trae.app/Contents/Resources/app/bin/marscode",
-  "Trae CN": "/Applications/Trae CN.app/Contents/Resources/app/bin/marscode",
-  VSCodium: "/Applications/VSCodium.app/Contents/Resources/app/bin/codium",
-  "VSCodium - Insiders": "/Applications/VSCodium - Insiders.app/Contents/Resources/app/bin/codium-insiders",
-  Windsurf: "/Applications/Windsurf.app/Contents/Resources/app/bin/windsurf",
+  Code: "code",
 };
 
 const cliPathsWindows: Record<string, string> = {
diff --git a/src/open-new-window.ts b/src/open-new-window.ts
index 834f52fed..ea43f8994 100644
--- a/src/open-new-window.ts
+++ b/src/open-new-window.ts
@@ -1,7 +1,8 @@
 import { Toast, closeMainWindow, showToast } from "@raycast/api";
-import { runAppleScript } from "@raycast/utils";
 import { build } from "./preferences";
 import { VSCodeBuild } from "./types";
+import { getVSCodeCLIFilename } from "./lib/vscode";
+import { runExec } from "./utils";
 
 /**
  * The index of the `New Window` menu item in the `File` menu.
@@ -35,27 +36,16 @@ const NewWindowMenuItemIndex: Record<VSCodeBuild, number> = {
  * We need to handle this case specially.
  */
 const makeNewWindow = async () => {
-  await runAppleScript(`
-    tell application "${build}"
-	    activate
-    end tell
-    delay(0.5)
-    tell application "${build}"
-	    activate
-    end tell
-
-    tell application "System Events"
-	    tell process "${build}"
-        tell menu bar 1
-          tell menu bar item 3
-            tell menu 1
-              click menu item ${NewWindowMenuItemIndex[build as VSCodeBuild] || 3}
-            end tell
-          end tell
-        end tell
-	    end tell
-    end tell
-  `);
+  const cliFilename = getVSCodeCLIFilename();
+  runExec([cliFilename], (error) => {
+    if (error) {
+      console.error(`Error opening new window: ${error}`);
+      showToast(Toast.Style.Failure, `Failed to open new window: ${error}`);
+      return;
+    } else {
+      showToast(Toast.Style.Success, `Opened in ${build}`);
+    }
+  });
 };
 
 export default async function command() {
